src/System.IO.FileSystem/tests/Enumeration/TrimmedPaths.netcoreapp.cs

```diff
@@ -47,18 +47,183 @@ public void TrimmedPathsDeletion_Windows()
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void TrimmedPathsText_Windows()
+        {
+            // Trailing spaces and periods are eaten when normalizing in Windows, making them impossible
+            // to access without using the \\?\ device syntax. We should, however, be able to open readers
+            // and writers from the the info class when enumerating.
+
+            DirectoryInfo directory = Directory.CreateDirectory(GetTestFilePath());
+            string fileOne = Path.Join(directory.FullName, "Trailing space ");
+            string fileTwo = Path.Join(directory.FullName, "Trailing period.");
+            File.WriteAllText(@"\\?\" + fileOne, "space");
+            File.WriteAllText(@"\\?\" + fileTwo, "period");
+
+            FileInfo[] files = directory.GetFiles();
+            Assert.Equal(2, files.Length);
+            foreach (FileInfo fi in directory.GetFiles())
+            {
+                using (StreamReader reader = fi.OpenText())
+                {
+                    string content = reader.ReadToEnd();
+                    if (fi.FullName.EndsWith(fileOne))
+                    {
+                        Assert.Equal("space", content);
+                    }
+                    else if (fi.FullName.EndsWith(fileTwo))
+                    {
+                        Assert.Equal("period", content);
+                    }
+                    else
+                    {
+                        Assert.False(true, $"Unexpected name '{fi.FullName}'");
+                    }
+                }
+
+                using (StreamWriter writer = fi.CreateText())
+                {
+                    writer.Write("foo");
+                }
+
+                using (StreamReader reader = fi.OpenText())
+                {
+                    Assert.Equal("foo", reader.ReadToEnd());
+                }
+
+                using (StreamWriter writer = fi.AppendText())
+                {
+                    writer.Write("bar");
+                }
+
+                using (StreamReader reader = fi.OpenText())
+                {
+                    Assert.Equal("foobar", reader.ReadToEnd());
+                }
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void TrimmedPathsCopyTo_Windows()
+        {
+            // Trailing spaces and periods are eaten when normalizing in Windows, making them impossible
+            // to access without using the \\?\ device syntax. We should, however, be able to copy them 
+            // without the special syntax from the info class when enumerating.
+
+            DirectoryInfo directory = Directory.CreateDirectory(GetTestFilePath());
+            string fileOne = Path.Join(directory.FullName, "Trailing space ");
+            string fileTwo = Path.Join(directory.FullName, "Trailing period.");
+            File.Create(@"\\?\" + fileOne).Dispose();
```

Comment:
why are we using dispose here ?

Reply:
It gives you a Filestream.

Reply:
To close the file handle right away so it doesn't stay locked past this point.